gaia/pysal_weights.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import pysal
+
+
+class PolygonCollection:
+    """
+    # function from:
+    # https://github.com/pysal/pysal/blob/master/pysal/weights/weights_from_geojson.ipynb
+    """
+    def __init__(self, polygons, bbox=None):
+        """
+
+        Parameters
+        ==========
+        polygons: dict
+                  key is polygon Id, value is PySAL Polygon object
+        bbox: list (optional)
+              [left, lower, right, upper]
+
+        Notes
+        =====
+        bbox is supported in geojson specification at both the feature and
+        feature collection level. However, not all geojson writers generate
+        the bbox at the feature collection level.
+
+        In those cases, the bbox property will be set on initial access.
+
+        """
+
+        self.type = pysal.cg.Polygon
+        self.n = len(polygons)
+        self.polygons = polygons
+        if bbox is None:
+            self._bbox = None
+        else:
+            self._bbox = bbox
+
+    @property
+    def bbox(self):
+        bboxes = np.array([self.polygons[p].bbox for p in self.polygons])
+        mins = bboxes.min(axis=0)
+        maxs = bboxes.max(axis=0)
+        self._bbox = [mins[0], mins[1], maxs[2], maxs[3]]
+        return self._bbox
+
+    def get(self, index):
+        return self.polygons[index]
+
+    def __getitem__(self, index):
+        return self.polygons[index]
+
+
+# converting gpd dataframes to polygon collections / points array
+def gpd_polygons(gpdf):
```

Comment:
@chuehlien I suggest we have conver utilies somewhat follow a api / set of standard methods. What you think?